makeStage2.sh

```diff
@@ -39,5 +40,66 @@ cd $usepop/pop/x/src/
 # ways to construct the Poplog system.
 # $usepop/pop/src/newpop -link -nox -norsv
 # $usepop/pop/src/newpop -link -x=-xm -norsv
+# $usepop/pop/src/newpop -link -x=-xt -norsv
+
+mkdir -p ${BUILD_HOME}/environments
+
+# We need to capture the Poplog environment for each build variant.
+echo_env() {
+    cmd='(usepop="'"$1"'" && . $usepop/pop/com/popenv.sh && env)'
+    env -i sh -c "$cmd" | sort | \
+    grep -v '^\(_\|SHLVL\|PWD\|poplib\|poplocal\(auto\|bin\)\?\)=' | \
+    sed -e 's!'"$1"'![//USEPOP//]!g'
+}
+
+### Xt #########################################################################
+
 $usepop/pop/src/newpop -link -x=-xt -norsv
 
+echo_env "$usepop" > ${BUILD_HOME}/environments/xt
+echo_env "$usepop/pop/.." > ${BUILD_HOME}/environments/xt-cmp
+( cd ${BUILD_HOME}/environments && \
+  sed -e 's!\[//USEPOP//]/pop/pop![//USEPOP//]/pop/pop-xt!g' < xt | \
+  sed -e 's!\[//USEPOP//]/pop/lib/psv/![//USEPOP//]/pop/lib/psv-xt/!g' > xt-new )
+
+mkdir -p $usepop/pop/pop-xt
+mkdir -p $usepop/pop/lib/psv-xt
+( cd $usepop/pop/pop; tar cf - . ) | ( cd $usepop/pop/pop-xt; tar xf - )
+( cd $usepop/pop/lib/psv; tar cf - . ) | ( cd $usepop/pop/lib/psv-xt; tar xf - )
+
+
+### motif ######################################################################
+
+$usepop/pop/src/newpop -link -x=-xm -norsv
+
+echo_env "$usepop" > ${BUILD_HOME}/environments/xm
+echo_env "$usepop/pop/.." > ${BUILD_HOME}/environments/xm-cmp
+( cd ${BUILD_HOME}/environments && \
+  sed -e 's!\[//USEPOP//]/pop/pop![//USEPOP//]/pop/pop-xm!g' < xm | \
+  sed -e 's!\[//USEPOP//]/pop/lib/psv/![//USEPOP//]/pop/lib/psv-xm/!g' > xm-new )
+
+mkdir -p $usepop/pop/pop-xm
+mkdir -p $usepop/pop/lib/psv-xm
+( cd $usepop/pop/pop; tar cf - . ) | ( cd $usepop/pop/pop-xm; tar xf - )
+( cd $usepop/pop/lib/psv; tar cf - . ) | ( cd $usepop/pop/lib/psv-xm; tar xf - )
+
+
+### nox ########################################################################
```

Comment:
Each of these 3 sections all follow the same code pattern. I think they should be extracted out to a parameterised helper function.

Reply:
Although this comment is marked as out of date it is still relevant. I might need a bit of help with this one.

Reply:
Alrighty, I have pushed the refactoring about as far as my skills go :)